subscription.go

```diff
@@ -344,8 +348,10 @@ func (s *Subscription) Close() error {
 		s.mu.Unlock()
 		return ErrSubscriptionClosed
 	}
-	s.mu.Unlock()
 	s.status = SUBCLOSED
+	s.mu.Unlock()
+
+	s.triggerUnsubscribeEvent()
```

Comment:
Will try to look a bit closer this weekend and merge if it works as intended. Thanks!